src/main/java/ladder/domain/ladder/Step.java

```diff
@@ -0,0 +1,84 @@
+package ladder.domain.ladder;
+
+import ladder.domain.init.LadderInitInfo;
+import ladder.domain.ladder.footstep.FootStepCreateStrategy;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+// 사다리 한단
+public class Step {
+    public static final int MINIMUM_LADDER_WIDTH = 1;
+
+    private final List<Column> columns;
+
+    private Step(final LadderInitInfo ladderInitInfo) {
+        validate(ladderInitInfo);
+
+        columns = new ArrayList<>(ladderInitInfo.getLadderWidth());
+        createHead(ladderInitInfo);
+        createMid(ladderInitInfo);
+        createTail(ladderInitInfo);
+    }
+
+    private void createTail(final LadderInitInfo ladderInitInfo) {
+        if (ladderInitInfo.getLadderWidth() > MINIMUM_LADDER_WIDTH) {
+            columns.add(createEndColumn());
+        }
+    }
+
+    private void createMid(final LadderInitInfo ladderInitInfo) {
+        if(ladderInitInfo.getLadderWidth() < MINIMUM_LADDER_WIDTH + 1) {
+            return;
+        }
+
+        Stream.generate(() -> createNextColumn(ladderInitInfo.getFootStepCreateStrategy()))
+                .limit(ladderInitInfo.getLadderWidth() - 2) //add except first and last column
+                .forEach(columns::add);
+    }
+
+    private void createHead(final LadderInitInfo ladderInitInfo) {
+        if (ladderInitInfo.getLadderWidth() == MINIMUM_LADDER_WIDTH) {
+            columns.add(Column.init(() -> false));
+            return;
+        }
+
+        columns.add(Column.init(ladderInitInfo.getFootStepCreateStrategy()));
+    }
+
+    private void validate(final LadderInitInfo ladderInitInfo) {
+        if (Objects.isNull(ladderInitInfo)) {
+            throw new IllegalArgumentException("Init info is null");
+        }
+
+        if (ladderInitInfo.getLadderWidth() < MINIMUM_LADDER_WIDTH) {
+            throw new IllegalArgumentException("Ladder width must larger or equal to " + MINIMUM_LADDER_WIDTH);
+        }
+    }
+
+    public static Step init(final LadderInitInfo ladderInitInfo) {
+        return new Step(ladderInitInfo);
+    }
+
+    public List<Boolean> toFootSteps() {
```

Comment:
이 메소드도 명확히 
```suggestion
    public List<Boolean> toRightFootSteps() {
```
로 하는건 어떨까요?